processor/metricstransformprocessor/aggregation_helper.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// timeseriesGroupByLabelValues is a data structure for grouping timeseries that will be aggregated
+type timeseriesGroupByLabelValues struct {
+	// key is a composite of the label values as a single string
+	// keyToTimeseriesMap groups timeseries by the label values
+	timeseries []*metricspb.TimeSeries
+	// keyToLabelValuesMap maps the keys to the actual label values objects
+	labelValues []*metricspb.LabelValue
+}
+
+// getLabelIdxs gets the indices of the labelSet labels' indices in the metric's descriptor's labels field
+// Returns the indices slice and a slice of the actual labels selected by this slice of indices
+func (mtp *metricsTransformProcessor) getLabelIdxs(metric *metricspb.Metric, labelSet map[string]bool) ([]int, []*metricspb.LabelKey) {
```

Comment:
Yea, I think this will be removed as well. Also based on the use cases for this function at the moment, I would consider this to be related to aggregating points as well (maybe not directly) because the points are aggregated together based on the label values located by the indices from this function.

Reply:
Maybe I should have said this file is to do with "merging timeseries". There is basically:
- Code to decide which timeseries to merge (i.e. divide timeseries into groups)
- Code to do merge timeseries together

I think it's best to keep those two things separate.

(in the new data structure replace timeseries with point & this will be even simpler)